systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaTemplates.java

```diff
@@ -227,6 +236,10 @@ private static KafkaBuilder defaultKafka(Kafka kafka, String name, int kafkaRepl
                     .endTopicOperator()
                 .endEntityOperator()
             .endSpec();
+        if (Environment.SYSTEM_TEST_STRIMZI_IMAGE_PULL_SECRET != null && !Environment.SYSTEM_TEST_STRIMZI_IMAGE_PULL_SECRET.isEmpty()) {
```

Comment:
Maybe a stupid question ... but why to use this and not configure the secret in the operator deployment directly? (using the `STRIMZI_IMAGE_PULL_SECRETS` env var which can configure secrets which will be passed to the operands?) It would seem cleaner than adding it to every custom resource. It would be IMHO also more stable because nobody would overwrite this in the ST itself by mistake.

Reply:
@scholzj I thought about passing the secret name with env using ContainerTemplate but I didn't think about passing the variable directly without manipulating the in-memory custom resource. Is that what you are suggesting? That sounds good.

Reply:
To CO deployment you can insert it to container env part. It will work for all Strimzi components except our clients, which are not handled by CO.

Reply:
That means, we modify the in-memory resource with ContainerTemplate's env part to add this variable. It was not clear to me if there was a way to pass the variable without modifying the in-memory resource.

Reply:
Thanks for the comments. I updated the code to use the env based image-pull-secret injection for kafka.